### DISCUSSION TITLE
Dynamic (Resource) ServiceName

### DISCUSSION THREAD

**Discussion by jonnepmyra** [open-telemetry/opentelemetry-dotnet#5691] [Q&A] [answered]
Hi!

I'm exporting traces from my application and it's registered like:

```
builder.Services.AddOpenTelemetry()
    .WithTracing(b =>
    {
      
           b.AddSource("someSource")
            .ConfigureResource((resource) =>
            {
                resource.AddService(serviceName: "SomeService");
            })
            .AddOtlpExporter((options) =>
            {
                options.Endpoint = new Uri("https://localhost:5341/ingest/otlp/v1/traces");
                options.Protocol = OtlpExportProtocol.HttpProtobuf;

            });
    });
```

...which makes the traces that end up in the trace endpoint to be "tagged" with `service.name = "SomeService"`

Would it be possible to somehow change the `serviceName` from "SomeService" to "service1" or "service2" depending on some metadata on the specific Activity that is being exported?

I'm actually trying to achieve multiple servicenames, within the same application and the servicename's are only known at runtime, when the trace/activity is started.

**Comment by cijothomas** [comment, marked as answer]
Not possible. Resource is immutable and once provider is built, you cannot change Resource.
There is some work in OTel spec about a [new concept of "Entities"](https://github.com/open-telemetry/community/pull/1976), which, when landed, might be what you need..... Sorry I don't have a better solution.

(Of course you can store service name as an attribute in Activity itself, instead of relying on Resource as a very hacky workaround)
